src/rpdk/core/contract/resource_client.py

```diff
@@ -361,22 +362,32 @@ def assert_time(self, start_time, end_time, action):
 
     @staticmethod
     def assert_primary_identifier(primary_identifier_paths, resource_model):
-        assert all(
-            traverse(resource_model, fragment_list(primary_identifier, "properties"))[0]
-            for primary_identifier in primary_identifier_paths
-        )
+        try:
+            assert all(
+                traverse(
+                    resource_model, fragment_list(primary_identifier, "properties")
+                )[0]
+                for primary_identifier in primary_identifier_paths
+            ), "Every model MUST include the primaryIdentifier"
+        except KeyError:
+            raise AssertionError("Every model MUST include the primaryIdentifier")
```

Comment:
can we throw key error instead with the same error message

Reply:
this assertion is aimed to make sure that primary identifier is returned within the `ResourceModel` object but in case it's missing or `null` we get `KeyError` instead of a failed assertion, which itself is extremely misleading. 

leaving `KeyError` will mean that the assertion message will never get thrown, bc it will be either a successful assertion (just if the key exists) or key error; 

I feel like `cli` with a testing component in it should not expose any unhandled runtime errors and instead surface what's causing incorrect behavior;